chart/templates/heartbeat.yaml

```diff
@@ -0,0 +1,42 @@
+{{with .Values -}}
+---
+###
+### Heartbeat
+###
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: linkerd-heartbeat
+  namespace: {{.Namespace}}
+  labels:
+    {{.ControllerComponentLabel}}: heartbeat
+    {{.ControllerNamespaceLabel}}: {{.Namespace}}
+  annotations:
+    {{.CreatedByAnnotation}}: {{.CliVersion}}
+spec:
+  schedule: "{{.HeartbeatSchedule}}"
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            {{.ControllerComponentLabel}}: heartbeat
+          annotations:
+            {{.CreatedByAnnotation}}: {{.CliVersion}}
+        spec:
+          serviceAccountName: linkerd-heartbeat
+          restartPolicy: OnFailure
+          containers:
+          - name: heartbeat
+            image: {{.ControllerImage}}
+            imagePullPolicy: {{.ImagePullPolicy}}
+            args:
+            - "heartbeat"
+            - "-prometheus-url=http://linkerd-prometheus.{{.Namespace}}.svc.cluster.local:9090"
+            - "-controller-namespace={{.Namespace}}"
+            - "-log-level={{.ControllerLogLevel}}"
+            {{- include "resources" .HeartbeatResources | indent 4 | trimPrefix "    " }}
```

Comment:
My go template foo is weak, can you explain this a bit?  why does it look different from the resource includes in other, uh.... resources?

Reply:
Yeah... this is an unfortunate side effect of how the [_resources.yaml template](https://github.com/linkerd/linkerd2/blob/f535c2d3d2f0da76bd62f3cbc1e39994a773ed21/chart/templates/_resources.yaml#L2) has a hard-coded indentation. This works for every other component because they are all under:
```yaml
spec:
  template:
    spec:
      containers:
      - resources:
```
...but the heartbeat cronjob is structured as:
```yaml
spec:
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - resources:
```
... the extra 2 levels motivates the `indent 4` (and then trimming the extra ws on the previous line)

a better fix would be to make `_resources.yaml` agnostic to indentation level, but i decided that was out of scope for this PR.

Reply:
You're looking for `nindent` - https://github.com/helm/helm/blob/master/docs/charts_tips_and_tricks.md#using-the-include-function

Reply:
filed #3119